FWCore/ParameterSet/python/Config.py

```diff
@@ -1165,6 +1165,25 @@ def toModify(self,obj, func=None,**kw):
     else:
       temp =_ParameterModifier(kw)
       temp(obj)
+  def toReplaceWith(self,toObj,fromObj):
+    """If the Modifier is chosen the internals of toObj will be associated with the internals of fromObj
+    """
+    if type(fromObj) != type(toObj):
+        raise TypeError("toReplaceWith requires both arguments to be the same class type")
+    if isinstance(fromObj,_ModuleSequenceType):
+        toObj._seq = fromObj._seq
+    elif isinstance(fromObj,_Parameterizable):
+        #clear old items just incase fromObj is not a complete superset of toObj
+        for p in toObj.parameterNames_():
+            delattr(toObj,p)
+        for p in fromObj.parameterNames_():
+            setattr(toObj,p,getattr(fromObj,p))
+        if isinstance(fromObj,_TypedParameterizable):
+            toObj._TypedParameterizable__type = fromObj._TypedParameterizable__type
+            
+    else:
+        raise TypeError("toReplaceWith does not work with type "+str(type(toObj)))
+
```

Comment:
I think you are right. I should have written a test showing it doesn't apply if not chosen.